lnwallet/wallet.go

```diff
@@ -184,6 +184,15 @@ type InitFundingReserveMsg struct {
 	// used.
 	ChanFunder chanfunding.Assembler
 
+	// AllowUtxoForFunding enables the channel funding workflow to restrict
+	// the selection of utxos when selecting the inputs for the channel
+	// opening. This does ONLY apply for the internal wallet backed channel
+	// opening case.
+	//
+	// NOTE: This is very useful when opening channels with unconfirmed
+	// inputs to make sure stable non-replaceable inputs are used.
+	AllowUtxoForFunding func(Utxo) bool
```

Comment:
hmm ok one style thing that stands out to me here is that we are making this a request specific field but it is not something that changes per request. So why not instead have this on the main `lnwallet.Config`?  Then it is just defined once and can be re-used over and over again per channel funding request.

Reply:
ie, to me, anything in `InitFundingReserveMsg` should be specific to that request

Reply:
if we wanna make things way more generic (so that this is a request specific thing) then how about making the field on `InitFundingReserveMsg` something like `UtxoFilter func(utxo) bool` and then pass that straight as a functional option to `btcwallet`. 

That way this does make sense on a per-request level here.

Reply:
i know there was potentially already some off-line discussion about this between other reviewers & the author - so apologies if this has already been covered - just want to make sure I understand the decision properly

Reply:
ok upon further inspection, I see that that would quite a bit of refactor since the lnwallet.Config gets create way before the sweeper is created. So perhaps not worth the effort to change this 👍

Reply:
I agree your proposal seems definitely better design wise, but I don't think it's worth the effort going down this route.